vite-project/src/strategypage/RadarChart.tsx

```diff
@@ -0,0 +1,266 @@
+
+      const currentPoint: point = { x: currentX, y: currentY };
+      statPoints[i] = currentPoint;
+    }
+  }
+
+  // function that draws inner shape and fills it in
+  function drawStatPolygon(
+    context: CanvasRenderingContext2D,
+    polygonPoints: point[],
+    fillStyle: string
+  ): void {
+    context.beginPath();
+    context.fillStyle = fillStyle;
+
+    for (let i = 0; i < inputs.length; i++) {
+      context.lineTo(statPoints[i].x, statPoints[i].y);
+    }
+
+    context.lineTo(statPoints[0].x, statPoints[0].y);
+
+    context.closePath();
+  }
+
+  //function that places the 10-100 digit
+  function placeStatDigits(
+    context: CanvasRenderingContext2D,
+    fillStyle: string,
+    strokeStyle: string
+  ): void {
+    const canvasRadius: number = context.canvas.width / 2;
+    const polygonRadius: number = canvasRadius * 0.75;
+
+    const digitPoint: point = {
+      x: polygonRadius + polygonRadius / 3.35,
+      y: polygonRadius + polygonRadius / 2.3,
+    };
+
+    context.strokeStyle = strokeStyle;
+    context.fillStyle = fillStyle;
+
+    for (let i = 1; i <= 10; i++) {
+      context.font = "15px Arial";
+      context.strokeText(String(i * 10), digitPoint.x, digitPoint.y);
+
+      context.fill();
+
+      context.font = "15px Arial";
+      context.fillText(String(i * 10), digitPoint.x, digitPoint.y);
+
+      context.fill();
+
+      digitPoint.y = digitPoint.y + polygonRadius / 10;
+      context.moveTo(digitPoint.x, digitPoint.y);
+    }
+  }
+
+  // draws the commponent
+  useEffect(() => {
+    const canvas = canvasRef.current;
+    if (!canvas) {
+      return;
+    }
+    const context = canvasRef.current.getContext("2d");
+    if (!context) {
+      return;
+    }
+
+    const canvasRadius: number = canvas.width / 2;
+    const polygonRadius: number = canvasRadius * 0.75;
+    context.clearRect(0, 0, canvas.width, canvas.height);
+
+    // draws the inner polygons
+
+    context.moveTo(canvasRadius, canvasRadius);
+
+    context.beginPath();
+    context.lineWidth = 1.5;
+
+    for (var i = 1; i < 10; i++) {
+      drawInnerPolygon((polygonRadius * i) / 10, canvas, "gray");
+      context.moveTo(canvasRadius, canvasRadius);
+    }
+
+    context.closePath();
+
+    //draws the outter polygon
+
+    context.beginPath();
+
+    context.lineWidth = 3;
+
+    drawOuterPolygon(polygonRadius, canvas, polygonPoints, "white");
+
+    context.closePath();
+
+    //puts lablse on the corners
+
+    context.beginPath();
+    context.font = "15px Arial";
+
+    drawLables(polygonRadius * 1.21, canvas);
+
+    context.closePath();
+    context.lineWidth = 3;
+    context.fill();
+    context.stroke();
+
+    //findes stat points
+
+    context.beginPath();
+
+    findeStatPoints(canvas, polygonPoints);
+
+    context.closePath();
+
+    // draws inner shape and fills it in
+
+    drawStatPolygon(context, polygonPoints, "rgb(96, 190, 235, 0.5)");
+
+    context.lineWidth = 3;
+    context.stroke();
+    context.fill();
```

Comment:
Why are some of the styles passed in as parameters and others are defined outside the function calls? This also applies to other functions, you need to decide whether your functions actually do the drawing or just place paths